core/src/storage/collections/btree_map/impls.rs

```diff
@@ -0,0 +1,1872 @@
+where
+    K: Encode + Decode + Ord,
+    V: Encode + Decode,
+{
+    /// Gets a reference to the key in the entry.
+    ///
+    /// # Examples
+    ///
+    /// ```no_compile
+    /// use ink_core::storage::BTreeMap;
+    ///
+    /// let mut map: BTreeMap<&str, usize> = new_btree_map();
+    /// map.entry("poneyland").or_insert(12);
+    /// assert_eq!(map.entry("poneyland").key(), &"poneyland");
+    /// ```
+    pub fn key(&self) -> &K {
+        self.tree
+            .get_kv(self.handle)
+            .expect("every occupied entry always has a key/value pair; qed")
+            .0
+    }
+
+    /// Gets a reference to the value in the entry.
+    ///
+    /// # Examples
+    ///
+    /// ```no_compile
+    /// use ink_core::storage::BTreeMap;
+    /// use ink_core::storage::btree_map::Entry;
+    ///
+    /// let mut map: BTreeMap<&str, usize> = new_btree_map();
+    /// map.entry("poneyland").or_insert(12);
+    ///
+    /// if let Entry::Occupied(o) = map.entry("poneyland") {
+    ///     assert_eq!(o.get(), &12);
+    /// }
+    /// ```
+    pub fn get(&self) -> &V {
+        self.tree
+            .get_kv(self.handle)
+            .expect("every occupied entry always has a key/value pair; qed")
+            .1
+    }
+
+    /// Gets a mutable reference to the value in the entry.
+    ///
+    /// # Examples
+    ///
+    /// ```no_compile
+    /// use ink_core::storage::BTreeMap;
+    /// use ink_core::storage::btree_map::Entry;
+    ///
+    /// let mut map: BTreeMap<&str, usize> = new_btree_map();
+    /// map.entry("poneyland").or_insert(12);
+    ///
+    /// assert_eq!(map["poneyland"], 12);
+    /// if let Entry::Occupied(mut o) = map.entry("poneyland") {
+    ///     *o.get_mut() += 10;
+    ///     assert_eq!(*o.get(), 22);
+    ///
+    ///     // We can use the same Entry multiple times.
+    ///     *o.get_mut() += 2;
+    /// }
+    /// assert_eq!(map["poneyland"], 24);
+    /// ```
+    pub fn get_mut(&mut self) -> &mut V {
+        self.kv_mut()
+            .expect("every occupied entry always has a key/value pair; qed")
+            .1
+    }
+
+    /// Converts the entry into a mutable reference to its value.
+    ///
+    /// If you need multiple references to the `OccupiedEntry`, see [`get_mut`].
+    ///
+    /// [`get_mut`]: #method.get_mut
+    ///
+    /// # Examples
+    ///
+    /// ```no_compile
+    /// use ink_core::storage::BTreeMap;
+    /// use ink_core::storage::btree_map::Entry;
+    ///
+    /// let mut map: BTreeMap<&str, usize> = new_btree_map();
+    /// map.entry("poneyland").or_insert(12);
+    ///
+    /// assert_eq!(map["poneyland"], 12);
+    /// if let Entry::Occupied(o) = map.entry("poneyland") {
+    ///     *o.into_mut() += 10;
+    /// }
+    /// assert_eq!(map["poneyland"], 22);
+    /// ```
+    pub fn into_mut(self) -> &'a mut V {
+        self.into_kv_mut()
+            .expect("every occupied entry always has a key/value pair; qed")
+            .1
```

Comment:
I'd love to see named structs instead of tuples or tuple structs in such cases since something like `.1` is not very readable tbh.